unsafe/src/main/java/org/apache/spark/unsafe/types/Interval.java

```diff
@@ -86,6 +86,31 @@ public Interval(int months, long microseconds) {
     this.microseconds = microseconds;
   }
 
+  public Interval doOp(Interval that, String op) {
```

Comment:
ok. I wanted to avoid defining codegen in two operations. Update later together.